native/src/widget/text_input.rs

```diff
@@ -1041,3 +841,182 @@ mod platform {
+        }
+
+        pub fn select_right(&mut self, value: &Value) {
+            if self.end < value.len() {
+                self.end += 1;
+            }
+        }
+
+        pub fn select_left_by_words(&mut self, value: &Value) {
+            self.end = value.previous_start_of_word(self.start);
+        }
+
+        pub fn select_right_by_words(&mut self, value: &Value) {
+            self.end = value.next_end_of_word(self.start);
+        }
+
+        pub fn select_all(&mut self, len: usize) {
+            self.start = 0;
+            self.end = len;
+        }
+        /* Selection section end */
+
+        /* Double/Triple click section */
+        // returns the amount of clicks on the same position in specific timeframe
+        // (1=double click, 2=triple click)
+        pub fn process_click(&mut self, position: Point) -> usize {
+            if position
+                == self.last_click_position.unwrap_or(Point { x: 0.0, y: 0.0 })
+                && self.click_count < 2
+                && SystemTime::now()
+                    .duration_since(
+                        self.last_click_timestamp
+                            .unwrap_or(SystemTime::UNIX_EPOCH),
+                    )
+                    .unwrap_or(Duration::from_secs(1))
+                    .as_millis()
+                    <= 500
+            {
+                self.click_count += 1;
+            } else {
+                self.click_count = 0;
+            }
+            self.last_click_position = Option::from(position);
+            self.last_click_timestamp = Option::from(SystemTime::now());
+            self.click_count
+        }
+        /* Double/Triple click section end */
+
+        /* "get info about cursor/selection" section */
+        pub fn is_selection(&self) -> bool {
+            self.start != self.end
+        }
+
+        // get start position of selection (can be left OR right boundary of selection)
+        pub(crate) fn start(&self) -> usize {
+            self.start
+        }
+
+        // get end position of selection (can be left OR right boundary of selection)
+        pub fn end(&self) -> usize {
+            self.end
+        }
+
+        // get left boundary of selection
+        pub fn left(&self) -> usize {
+            self.start.min(self.end)
+        }
+
+        // get right boundary of selection
+        pub fn right(&self) -> usize {
+            self.start.max(self.end)
+        }
+
+        pub fn cursor_position(&self, value: &Value) -> (usize, usize) {
+            (self.start.min(value.len()), self.end.min(value.len()))
+        }
+
+        pub fn cursor_position_left(&self, value: &Value) -> usize {
+            let (a, b) = self.cursor_position(value);
+            a.min(b)
+        }
+
+        pub fn cursor_position_right(&self, value: &Value) -> usize {
+            let (a, b) = self.cursor_position(value);
+            a.max(b)
+        }
+
+        pub fn draw_position(&self, value: &Value) -> usize {
```

Comment:
It doesn't make too much sense to talk about drawing in this context. We should find a better name.

Reply:
cleanup section again. I chose this name just to understand what the renderer wants (based on Finnerales changes). So yes, i want to rename it too or maybe even replace/remove it somehow (because its just used by renderer/widget/text_input).